packages/api-bindings/src/lens/utils/CollectModuleSettings.ts

```diff
@@ -103,10 +132,30 @@ function isMultirecipientCollectModuleSettings(
 }
 
 function buildCollectFee(
-  module: CollectModuleSettings,
+  module: gql.ProtocolSharedRevenueCollectOpenActionSettings,
+): CollectFee | undefined;
+function buildCollectFee(
+  module: gql.LegacySimpleCollectModuleSettings | gql.SimpleCollectOpenActionSettings,
+): CollectFee | undefined;
+function buildCollectFee(
+  module:
+    | gql.LegacyMultirecipientFeeCollectModuleSettings
+    | gql.MultirecipientFeeCollectOpenActionSettings,
+): MultirecipientCollectFee;
+function buildCollectFee(
+  module: gql.LegacyFeeCollectModuleSettings | gql.LegacyTimedFeeCollectModuleSettings,
+): CollectFee;
+function buildCollectFee(
+  module: gql.LegacyAaveFeeCollectModuleSettings | gql.LegacyErc4626FeeCollectModuleSettings,
+): CollectFee;
+function buildCollectFee(
+  module:
+    | gql.LegacyLimitedFeeCollectModuleSettings
+    | gql.LegacyLimitedTimedFeeCollectModuleSettings,
+): CollectFee;
+function buildCollectFee(
+  module: Extract<CollectModuleSettings, { amount: gql.Amount }>,
```

Comment:
does it read better than `module: CollectModuleSettings` ? or is the type safety better this way?

Reply:
I changed it so that works only with modules that can have a fee.

This was more type safe at the call site.

I also provided the overloads so that call site is even more accurate as I experienced a typo while I was working and wasn't flagging the call.

Finally the implementation signature just supports all possible overrides.